### ISSUE TITLE
`<CartesianGrid />` sets `offset` attribute to `[object Object]`

### ISSUE THREAD

**Issue by EvHaus** [recharts/recharts#3810]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://recharts.org/en-US/api/Legend](https://recharts.org/en-US/api/Legend)

### Steps to reproduce
It seems like all `<CartesianGrid />` elements that recharts renders have their DOM elements set the `offset` value to the string `[object Object]`. You can reproduce this by going to the official documentation at https://recharts.org/en-US/api/Legend and inspect the DOM. You'll see:

```
<g class="recharts-cartesian-grid-horizontal"><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="80" y="41" width="620" height="174" offset="[object Object]" x1="80" y1="215" x2="700" y2="215"></line><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="80" y="41" width="620" height="174" offset="[object Object]" x1="80" y1="171.5" x2="700" y2="171.5"></line><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="80" y="41" width="620" height="174" offset="[object Object]" x1="80" y1="128" x2="700" y2="128"></line><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="80" y="41" width="620" height="174" offset="[object Object]" x1="80" y1="84.5" x2="700" y2="84.5"></line><line stroke-dasharray="3 3" stroke="#ccc" fill="none" x="80" y="41" width="620" height="174" offset="[object Object]" x1="80" y1="41" x2="700" y2="41"></line></g>
```

### What is expected?
Since the bug doesn't seem to affect the actual rendering of the element, it would be nice to keep the DOM output clean and simply remove the "offset" attribute entirely.

### What is actually happening?
DOM nodes are rendered with an invalid `offset="[object Object]"` attribute value

| Environment | Info |
|---|---|
| Recharts | v2.8.0 |
| React | 18.2.0 |
| System | Linux |
| Browser | Chrome |

**Comment by ckifer** [comment]
Should be an easy enough one to fix, feel free to contribute!

**Comment by ckifer** [comment]
resolved in 2.9!
